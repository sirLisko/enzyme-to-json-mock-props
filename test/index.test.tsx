import * as React from "react";
import { shallow, CommonWrapper } from "enzyme";

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import mockProps from "../src";

const SubComponent = () => <div />;

const Component = (props: any) => <SubComponent {...props} />;

describe("enzyme-to-json-mock-props", () => {
  let wrapper: CommonWrapper;
  const props = {
    foo: {
      bar: 123
    },
    asd: {
      qwe: 321
    },
    iop: "asd"
  };

  beforeEach(() => {
    wrapper = shallow(<Component {...props} />);
  });

  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should mock the indicated props", () => {
    expect(mockProps(wrapper, ["foo", "poi", "iop"])).toMatchSnapshot();
  });
});
