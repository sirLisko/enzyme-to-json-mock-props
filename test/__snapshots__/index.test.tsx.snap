// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme-to-json-mock-props should mock the indicated props 1`] = `
<SubComponent
  asd={
    Object {
      "qwe": 321,
    }
  }
  foo="[foo]"
  iop="[iop]"
/>
`;

exports[`enzyme-to-json-mock-props should render properly 1`] = `
<SubComponent
  asd={
    Object {
      "qwe": 321,
    }
  }
  foo={
    Object {
      "bar": 123,
    }
  }
  iop="asd"
/>
`;
